File: sujeong/최소직사각형.swift
import Foundation

// 최초풀이
func solution(_ sizes:[[Int]]) -> Int {
    
    var maxOfMaxLength: Int = 0
    var maxOfMinLength: Int = 0
    
     for size in sizes {
         let maxLength = max(size[0], size[1])
         let minLength = min(size[0], size[1])
         if maxLength > maxOfMaxLength {
             maxOfMaxLength = maxLength
         }
         if minLength > maxOfMinLength {
             maxOfMinLength = minLength
         }
     }
    
    return maxOfMaxLength * maxOfMinLength
}

// 개선 풀이1
func solution2(_ sizes:[[Int]]) -> Int {
    
    let maxOfMaxLength: Int = sizes.map{ $0.max() ?? 0 }.max() ?? 0
    let maxOfMinLength: Int = sizes.map{ $0.min() ?? 0 }.max() ?? 0
    
    return maxOfMaxLength * maxOfMinLength
}

// 간략 코드
func solution3(_ sizes:[[Int]]) -> Int {
    return (sizes.map{ $0.max() ?? 0 }.max() ?? 0) * (sizes.map{ $0.min() ?? 0 }.max() ?? 0)
}

File: seungmi/week1/최소직사각형.py

def solution(sizes):
    max_w=0
    max_h=0
    
    for size in sizes:
      #세로의 길이가 더 길 때
        if size[0]<size[1] :
            max_w=max(size[1],max_w)
            max_h=max(size[0],max_h)
      #가로의 길이가 더 길 때
        else:
            max_w=max(size[0],max_w)
            max_h=max(size[1],max_h)
    
    return  max_w*max_h